src/all/namicomi/src/eu/kanade/tachiyomi/extension/all/namicomi/dto/EntityDto.kt

```diff
@@ -0,0 +1,16 @@
+package eu.kanade.tachiyomi.extension.all.namicomi.dto
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+abstract class EntityDto {
+    val id: String = ""
+    val relationships: List<EntityDto> = emptyList()
+    abstract val attributes: AttributesDto?
+}
+
+@Serializable
+abstract class AttributesDto
```

Comment:
```suggestion
@Serializable
sealed class EntityDto {
    val id: String = ""
    val relationships: List<EntityDto> = emptyList()
    abstract val attributes: AttributesDto?
}

@Serializable
sealed interface AttributesDto
```
And they will automatically become polymorphic. You still need to keep the default deserializer tho.